OCCommunicationLib/OCCommunicationLib/OCCommunication.m

```diff
@@ -776,6 +777,73 @@ - (void) readSharedByServer: (NSString *) serverPath andPath: (NSString *) path
     }];
 }
 
+- (void) shareFileOrFolderByServerPath:(NSString *)serverPath
+                  withFileOrFolderPath:(NSString *)filePath
+                              password:(NSString *)password
+                        expirationTime:(NSString *)expirationTime
+                          publicUpload:(BOOL)publicUpload
+                              linkName:(NSString *)linkName
+                       onCommunication:(OCCommunication *)sharedOCCommunication
+                        successRequest:(void(^)(NSHTTPURLResponse *response, NSString *token, NSString *redirectedServer)) successRequest
+                        failureRequest:(void(^)(NSHTTPURLResponse *response, NSError *error, NSString *redirectedServer)) failureRequest {
+    
+    serverPath = [serverPath encodeString:NSUTF8StringEncoding];
+    serverPath = [serverPath stringByAppendingString:k_url_acces_shared_api];
+    
+    OCWebDAVClient *request = [OCWebDAVClient new];
+    request = [self getRequestWithCredentials:request];
+    
+    
+    [request shareByLinkFileOrFolderByServer:serverPath andPath:filePath password:password expirationTime:expirationTime publicUpload:publicUpload linkName:linkName
+                             onCommunication:sharedOCCommunication success:^(NSHTTPURLResponse *response, id responseObject) {
+        NSData *responseData = (NSData*) responseObject;
+        
+        OCXMLShareByLinkParser *parser = [[OCXMLShareByLinkParser alloc]init];
```

Comment:
For the future: all the shares responses should use the same *parser, returning a Share object with values not received in the response in a default state.